### PR TITLE
JSON input IDD changes only

### PR DESCRIPTION
**Pull request by mbadams5**
Pull request overview
---------------------
These are the IDD changes for #5893.

I don't know if any of these changes require a transition or not. The only one potentially is `MutuallyExlcusive` to `MutuallyExclusive `.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by mjwitte** [comment]
We *could* add a transition rule for good measure, but the code is checking a literal string [MUTUALLYEXCLUSIVE](https://github.com/NREL/EnergyPlus/blob/2dd86cbb4d3da5d3c1fade4d8b478d627005060d/src/EnergyPlus/WaterThermalTanks.cc#L2389-L2393) rather than using an input processor key check, so anything with `MutuallyExlcusive` would fail.  So, I'd say no transition necessary.

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -70103,7 +70108,7 @@ WaterHeater:HeatPump:PumpedCondenser,
        \note proportional to the schedule value (i.e. 0.4 = 40% outdoor air, 60% zone air).
   A29, \field Tank Element Control Logic
        \type choice
-       \key MutuallyExlcusive
+       \key MutuallyExclusive
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:70111]: ~~The [code](https://github.com/NREL/EnergyPlus/blob/2dd86cbb4d3da5d3c1fade4d8b478d627005060d/src/EnergyPlus/WaterThermalTanks.cc#L2391) is expecting MutuallyExclusive, so we should change any files that inadvertently used this wrong key choice.~~

**Review by Myoldmopar — COMMENTED** [review]
This looks great.  I will add transition rule and code for the typo, but afterwards this will drop right in.

File: idd/Energy+.idd.in (hunk 1 above)
Comment (Myoldmopar) [reply idd/Energy+.idd.in:70111]: SCRATCH THAT!  👍

**Comment by Myoldmopar** [comment]
OK, since no transition is necessary, this is ready.  I'll wait until Marvin gets around to seeing the IDD and using it in some integration tests.  If he is happy, the IDD is fine, and this will go right in.  Thanks for breaking this from the rest of the work, @mbadams5.

**Comment by Myoldmopar** [comment]
Thanks @mbadams5 I'm merging this now.

**Pull request merged**